spatialdata/_core/transform.py

```diff
@@ -1,86 +1,526 @@
+            assert ndim is not None
+            scale = np.ones(ndim, dtype=float)
+
+        if ndim is None:
+            assert scale is not None
+            if isinstance(scale, list):
+                scale = np.array(scale)
+            ndim = len(scale)
+
+        assert type(scale) == np.ndarray
+        assert len(scale) == ndim
+        self.scale = scale
+        self._ndim = ndim
+
+    @property
+    def ndim(self) -> Optional[int]:
+        return self._ndim
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            "type": "scale",
+            "scale": self.scale.tolist(),
+        }
+
+    def transform_points(self, points: ArrayLike) -> ArrayLike:
+        return points * self.scale
+
+    def inverse(self) -> BaseTransformation:
+        new_scale = np.zeros_like(self.scale)
+        new_scale[np.nonzero(self.scale)] = 1 / self.scale[np.nonzero(self.scale)]
+        return Scale(scale=new_scale)
+
+    def to_affine(self) -> Affine:
+        assert self.ndim == 2
+        m: ArrayLike = np.vstack((np.hstack((np.diag(self.scale), np.zeros((2, 1)))), np.array([0.0, 0.0, 1.0])))
+        return Affine(affine=m)
 
-        if scale_factors is None:
-            if translation is not None:
-                scale_factors = np.ones(len(translation), dtype=float)
-            elif ndim is not None:
-                scale_factors = np.ones(ndim, dtype=float)
+
+class Affine(BaseTransformation):
+    def __init__(
+        self,
+        affine: Optional[Union[ArrayLike, List[Any]]] = None,
+    ) -> None:
+        """
+        class for storing scale transformations.
+        """
+        if affine is None:
+            affine = np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]], dtype=float)
+        else:
+            if isinstance(affine, list):
+                affine = np.array(affine)
+            if len(affine.shape) == 1:
+                assert len(affine) == 6
+                affine = np.vstack([affine.reshape((2, 3)), np.array([0.0, 0.0, 1.0])])
+            assert affine.shape == (3, 3)
+
+        self.affine = affine
+        self._ndim = 2
+
+    @property
+    def ndim(self) -> Optional[int]:
+        return self._ndim
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            "type": "affine",
+            "affine": self.affine[:2, :].ravel().tolist(),
+        }
+
+    def transform_points(self, points: ArrayLike) -> ArrayLike:
+        p = np.vstack([points.T, np.ones(points.shape[0])])
+        q = self.affine @ p
+        return q[:2, :].T  # type: ignore[no-any-return]
+
+    def inverse(self) -> BaseTransformation:
+        # naive check for numerical stability
+        a = self.affine[0, 0]
+        b = self.affine[0, 1]
+        m = self.affine[0, 2]
+        c = self.affine[1, 0]
+        d = self.affine[1, 1]
+        n = self.affine[1, 2]
+        det = a * d - b * c
+        closed_form = np.array([[d, -c, 0], [-b, a, 0], [b * n - d * m, c * m - a * n, det]])
```

Comment:
remember you explaining this, I wonder if it makes sense to have it as separate function called by the method and documented. Also if there is no already something like this in scipy?

Reply:
Good point. For the ndim case we'll have to use something like `numpy.linalg.inv` anyway. I'll modify the code.